libafl/src/corpus/inmemory_ondisk.rs

```diff
@@ -259,19 +293,21 @@ where
                 return Ok(());
             }
 
-            let new_lock_filename = format!(".{new_filename}.lafl_lock");
+            if self.locking {
```

Comment:
Why would you not be locking? Are you using one folder per node?

Reply:
If you aren't forking, you probably don't want to lock.

Reply:
fork as in multicore? Then I agree. Although you can also restart, you wouldn't have to fork :P

Reply:
Does libafl_libfuzzer support a non-forking multiprocess mode btw? Might be relevant for some targets

Reply:
> Does libafl_libfuzzer support a non-forking multiprocess mode btw? Might be relevant for some targets

How do you mean? With e.g. threads instead of forks?

Reply:
LibAFL launcher can start new processes instead of fork, yes

Reply:
So, it should just use whatever's appropriate for the system. I don't specify fork/process explicitly.

Reply:
You need to specifically not use fork for some targets like browsers - they don't cope well with forking since all vital threads die in the forked process.